jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildService.java

```diff
@@ -134,9 +136,22 @@ static ImageConfiguration prependRegistry(ImageConfiguration imageConfiguration,
             imageConfiguration.setName(registry + "/" + imageConfiguration.getName());
             imageConfiguration.setRegistry(registry);
         }
+        prependRegistryForFromImage(imageConfiguration, registry);
         return imageConfiguration;
     }
 
+    private static ImageConfiguration prependRegistryForFromImage(ImageConfiguration imageConfig, String registry) {
+        BuildConfiguration buildConfiguration = imageConfig.getBuildConfiguration();
+        String fromImage = BuildUtil.extractBaseFromConfiguration(buildConfiguration);
+        String imageRegistry = fromImage != null ? new ImageName(fromImage).getRegistry() : null;
+        final String actualRegistry = EnvUtil.firstRegistryOf(imageRegistry, registry, imageConfig.getRegistry());
```

Comment:
Maybe not execute this if imageRegistry not null?

Reply:
m? Why? We get list of potential registry sources. Then we get 1st from the query that is valid. As common logic for this plugin.

Reply:
`actualRegistry` is used in the next line if `imageRegistry` is null. So if `imageRegistry` is not null, evaluating `actualRegistry` is useless as it won't be used afterward.

Reply:
You are absolutely right, thank you.
Fixed (removed unnecessary condition).